exporters/prometheus/src/exporter_utils.cc

```diff
@@ -0,0 +1,328 @@
+ */
+std::vector<prometheus_client::MetricFamily> PrometheusExporterUtils::TranslateToPrometheus(
+    const std::vector<std::unique_ptr<sdk::metrics::ResourceMetrics>> &data)
+{
+  if (data.empty())
+  {
+    return {};
+  }
+
+  // initialize output vector
+  std::vector<prometheus_client::MetricFamily> output(data.size());
+
+  // iterate through the vector and set result data into it
+  int i = 0;
+  for (const auto &r : data)
+  {
+    SetMetricFamily(*r, &output[i]);
+    i++;
+  }
+
+  return output;
+}
+
+/**
+ * Set value to metric family according to record
+ */
+void PrometheusExporterUtils::SetMetricFamily(sdk::metrics::ResourceMetrics &data,
+                                              prometheus_client::MetricFamily *metric_family)
+{
+  SetMetricFamilyByAggregator(data, "", metric_family);
+}
+
+/**
+ * Sanitize the given metric name or label according to Prometheus rule.
+ *
+ * This function is needed because names in OpenTelemetry can contain
+ * alphanumeric characters, '_', '.', and '-', whereas in Prometheus the
+ * name should only contain alphanumeric characters and '_'.
+ */
+std::string PrometheusExporterUtils::SanitizeNames(std::string name)
+{
+  // replace all '.' and '-' with '_'
+  std::replace(name.begin(), name.end(), '.', '_');
+  std::replace(name.begin(), name.end(), '-', '_');
+
+  return name;
+}
+
+/**
+ * Set value to metric family for different aggregator
+ */
+void PrometheusExporterUtils::SetMetricFamilyByAggregator(
+    const sdk::metrics::ResourceMetrics &data,
+    std::string labels_str,
+    prometheus_client::MetricFamily *metric_family)
+{
+  for (const auto &instrumentation_info : data.instrumentation_info_metric_data_)
+  {
+    auto origin_name    = instrumentation_info.instrumentation_library_->GetName();
+    auto sanitized      = SanitizeNames(origin_name);
+    metric_family->name = sanitized;
+    for (const auto &metric_data : instrumentation_info.metric_data_)
+    {
+      auto time = metric_data.start_ts.time_since_epoch();
+      for (const auto &point_data_attr : metric_data.point_data_attr_)
+      {
+        auto kind                                = getAggregationType(point_data_attr.point_data);
+        const prometheus_client::MetricType type = TranslateType(kind);
+        metric_family->type                      = type;
+        if (type == prometheus_client::MetricType::Histogram)  // Histogram
+        {
+          auto histogram_point_data =
+              nostd::get<sdk::metrics::HistogramPointData>(point_data_attr.point_data);
+          auto boundaries = histogram_point_data.boundaries_;
+          auto counts     = histogram_point_data.counts_;
+          SetData(std::vector<double>{nostd::get<double>(histogram_point_data.sum_),
+                                      (double)histogram_point_data.count_},
+                  boundaries, counts, labels_str, time, metric_family);
+        }
+        else  // Counter, Untyped
```

Comment:
We can also add for LastValue/Gauge ?

Reply:
What's the conversion type for `LastValue`?